### DISCUSSION TITLE
Counterbalancing

### DISCUSSION THREAD

**Discussion by au-justin** [jspsych/jsPsych#2675] [General]
I am trying to set up a counterbalanced experiment and am not sure how to do this on the Gorilla online experiment platform. I would like every participant to see the same images but with the button choices flipped such that there is a balanced sample.

Is there a simple method to implement this directly in the code or would I need to use a database? If so, how might this work with online platforms? 

Additionally, if I would like to use responses during the experiment to inform a subsequent component of the task, how might this be implemented in the code? For example, if participants choose 'one' and 'twenty', how can half the people be shown these choices and half be given 'two' and 'ten' (the choice they didn't select)? 

```javascript 
var test_stimuli = [
        {stimulus: images[0], choices: ['one', 'two']}, // 1
        {stimulus: images[1], choices: ['ten', 'twenty']}, // 2
];

var test = {
      type: jsPsychImageButtonResponse,
      stimulus: jsPsych.timelineVariable('stimulus'),
     choices: jsPsych.timelineVariable('choices'),
button_html: ['<button class="jspsych-btn" style = "position: relative; left: -50px; top: 100px; font-size: 40px;           height: 200px; width: 250px">%choice%</button>',
                    '<button class="jspsych-btn" style = "position: relative; left: 50px; top: 100px; font-size: 40px; height: 200px; width: 250px">%choice%</button>',
                    ],
data: {
       task: 'response'
     }
```

Thank you!

**Comment by cratlagh** [comment]
Hi,

For the first part of your question you'd need to do some stuff on the server-side. I've had some success with counter-balancing using the JATOS server 'Batch Session' approach.

See the example study 'Randomize Tasks Between Workers' for an example of how this is done:

[https://www.jatos.org/Example-Studies.html](https://www.jatos.org/Example-Studies.html)

Direct link to the example file download here as the page doesn't seem to be consistently loading at the moment:

[Example files](https://github.com/JATOS/JATOS_examples/raw/master/examples/randomize_tasks_between_workers.jzip)

Regarding your 2nd query, maybe look into loop nodes. I'm not sure if they will help with this but it's start. I've used them for repeating practice trials when responses are incorrect etc. There may be a way to use them with conditionals to achieve what you have in mind.

Cheers.

**Reply by au-justin** [reply]
Thank you @cratlagh for the suggestions! 

I'm not sure how useful the JATOS examples are since I'm using Gorilla. Does anyone know how to counterbalance in Gorilla with jsPsych code? I would appreciate any ideas in this regard and if someone could provide a worked out solution that would be even better.
